controllers/csi/metadata/sqlite.go

```diff
@@ -0,0 +1,308 @@
+package metadata
+
+import (
+	"database/sql"
+	"fmt"
+	"path/filepath"
+
+	dtcsi "github.com/Dynatrace/dynatrace-operator/controllers/csi"
+	"github.com/Dynatrace/dynatrace-operator/logger"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	sqliteDriverName = "sqlite3"
+
+	tenantsTableName       = "tenants"
+	tenantsCreateStatement = `
+	CREATE TABLE IF NOT EXISTS tenants (
+		UUID VARCHAR NOT NULL,
+		LatestVersion VARCHAR NOT NULL,
+		Dynakube VARCHAR NOT NULL,
+		PRIMARY KEY (UUID)
+	); `
+
+	volumesTableName       = "volumes"
+	volumesCreateStatement = `
+	CREATE TABLE IF NOT EXISTS volumes (
+		ID VARCHAR NOT NULL,
+		PodName VARCHAR NOT NULL,
+		Version VARCHAR NOT NULL,
+		TenantUUID VARCHAR NOT NULL,
+		PRIMARY KEY (ID)
+	);`
+
+	insertTenantStatement = `
+	INSERT INTO tenants (UUID, LatestVersion, Dynakube)
+	VALUES (?,?,?);
+	`
+	updateTenantStatement = `
+	UPDATE tenants
+	SET LatestVersion = ?, Dynakube = ?
+	WHERE UUID = ?;
+	`
+
+	getTenantStatement = `
+	SELECT LatestVersion, Dynakube
+	FROM tenants
+	WHERE UUID = ?;
+	`
+
+	getTenantViaDynakubeStatement = `
+	SELECT UUID, LatestVersion
+	FROM tenants
+	WHERE Dynakube = ?;
+	`
+
+	insertVolumeStatement = `
+	INSERT INTO volumes (ID, PodName, Version, TenantUUID)
+	VALUES (?,?,?,?);
+	`
+
+	getVolumeStatement = `
+	SELECT PodName, Version, TenantUUID
+	FROM volumes
+	WHERE ID = ?;
+	`
+
+	deleteVolumeStatement = "DELETE FROM volumes WHERE ID = ?;"
+
+	deleteTenantStatement = "DELETE FROM tenants WHERE UUID = ?;"
+
+	getUsedVersionsStatement = `
+	SELECT Version
+	FROM volumes
+	WHERE TenantUUID = ?;
+	`
+
+	getPodNamesStatement = `
+	SELECT ID, PodName
+	FROM volumes;
+	`
+
+	getDynakubesStatement = `
+	SELECT UUID, Dynakube
+	FROM tenants;
+	`
+)
+
+var (
+	log    = logger.NewDTLogger().WithName("storage")
+	dbPath = filepath.Join(dtcsi.DataPath, "csi.db")
```

Comment:
Overwriting this variable in tests is fragile. Would make it an argument to the constructor.